hudi-metastore/src/main/java/org/apache/hudi/common/table/HoodieTableMetastoreClient.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.hudi.common.config.HoodieMetastoreConfig;
+import org.apache.hudi.common.fs.ConsistencyGuardConfig;
+import org.apache.hudi.common.fs.FileSystemRetryConfig;
+import org.apache.hudi.common.model.HoodieTableType;
+import org.apache.hudi.common.table.timeline.HoodieActiveTimeline;
+import org.apache.hudi.common.table.timeline.HoodieInstant;
+import org.apache.hudi.common.table.timeline.versioning.TimelineLayoutVersion;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.config.HoodieWriteConfig;
+import org.apache.hudi.exception.HoodieException;
+import org.apache.hudi.metastore.client.HoodieMetastoreClient;
+import org.apache.hudi.metastore.client.RetryingHoodieMetastoreClient;
+import org.apache.hudi.metastore.thrift.NoSuchObjectException;
+import org.apache.hudi.metastore.thrift.Table;
+import org.apache.hudi.common.table.timeline.HoodieMetastoreBasedTimeline;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+import java.util.List;
+import java.util.Set;
+
+public class HoodieTableMetastoreClient extends HoodieTableMetaClient {
+  private static final Logger LOG = LogManager.getLogger(HoodieTableMetastoreClient.class);
+
+  private String databaseName;
+  private String tableName;
+  private Table table;
+  private HoodieMetastoreClient metastoreClient;
+
+  public HoodieTableMetastoreClient(Configuration conf, ConsistencyGuardConfig consistencyGuardConfig, FileSystemRetryConfig fileSystemRetryConfig,
+      String databaseName, String tableName, HoodieMetastoreConfig config) {
+    super(conf, config.getString(HoodieWriteConfig.BASE_PATH), false, consistencyGuardConfig, Option.of(TimelineLayoutVersion.CURR_LAYOUT_VERSION),
+        config.getString(HoodieTableConfig.PAYLOAD_CLASS_NAME), fileSystemRetryConfig);
+    if (databaseName == null || tableName == null) {
+      throw new HoodieException("The database and table name have to be specified");
+    }
+    this.databaseName = databaseName;
+    this.tableName = tableName;
+    this.metastoreConfig = config;
+    this.metastoreClient = RetryingHoodieMetastoreClient.getProxy(config);
+    this.table = initOrGetTable(databaseName, tableName, config);
+    // TODO: transfer table parameters to table config
+    this.tableConfig = new HoodieTableConfig();
+    tableConfig.setTableVersion(HoodieTableVersion.THREE);
+    tableConfig.setAll(config.getProps());
+  }
+
+  private Table initOrGetTable(String db, String tb, HoodieMetastoreConfig config) {
+    Table table;
+    try {
+      table = metastoreClient.getTable(databaseName, tableName);
+    } catch (HoodieException e) {
+      if (e.getCause() instanceof NoSuchObjectException) {
+        LOG.info(String.format("Table %s.%s doesn't exist, will create it.", db, tb));
+        table = new Table();
+        table.setDbName(db);
+        table.setTableName(tb);
+        table.setLocation(config.getString(HoodieWriteConfig.BASE_PATH));
+        table.setOwner("");
```

Comment:
is it possible to set owner  from get UGI ?

Reply:
Cloud u explain it?

Reply:
UserGroupInformation.getCurrentUser.getShortUserName

Reply:
UserGroupInformation.getCurrentUser.getShortUserName

Reply:
fix